handler.go

```diff
@@ -71,8 +71,8 @@ func NewUnaryHandler[Req, Res any](
 		if err != nil {
 			return err
 		}
-		mergeHeaders(conn.ResponseHeader(), response.Header())
-		mergeHeaders(conn.ResponseTrailer(), response.Trailer())
+		mergeMetadataHeaders(conn.ResponseHeader(), response.Header())
+		mergeMetadataHeaders(conn.ResponseTrailer(), response.Trailer())
```

Comment:
This feels a bit brittle. What is the different between `mergeHeaders` and `mergeMetadataHeaders`? Maybe they should be renamed so the difference and their purpose is more clear? The doc comment added to `mergeHeaders` isn't very helpful.

Reply:
Used `mergeNonProtocolHeaders` as an alt.